bitnami/scylladb/templates/_helpers.tpl

```diff
@@ -0,0 +1,295 @@
+{{- $replicaCount := int .Values.replicaCount }}
+{{- $seedCount := int .Values.cluster.seedCount }}
+{{- if or (lt $seedCount 1) (gt $seedCount $replicaCount) }}
+scylladb: cluster.seedCount
+
+    Number of seed nodes must be greater or equal than 1 and less or
+    equal to `replicaCount`.
+{{- end -}}
+{{- end -}}
+
+{{/* Validate values of Scylladb - Tls enabled */}}
+{{- define "scylladb.validateValues.tls" -}}
+{{- if and (include "scylladb.tlsEncryptionEnabled" .) (not .Values.tls.autoGenerated.enabled) (not .Values.tls.existingSecret) }}
+scylladb: tls.enabled
+    In order to enable TLS, you also need to provide
+    an existing secret containing the certificate/keyfile or
+    enable auto-generated certificates.
+{{- end -}}
+{{- end -}}
+
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Return  the proper Commit Storage Class
+{{ include "scylladb.commitstorage.class" ( dict "persistence" .Values.path.to.the.persistence "global" $) }}
+*/}}
+{{- define "scylladb.commitstorage.class" -}}
+{{- $storageClass := .persistence.commitStorageClass -}}
+{{- if .global -}}
+    {{- if .global.storageClass -}}
+        {{- $storageClass = .global.commitStorageClass -}}
+    {{- end -}}
+{{- end -}}
+
+{{- if $storageClass -}}
+  {{- if (eq "-" $storageClass) -}}
+      {{- printf "storageClassName: \"\"" -}}
+  {{- else }}
+      {{- printf "storageClassName: %s" $storageClass -}}
+  {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return type of internode communication connections should be configured
+*/}}
+{{- define "scylladb.internode.tlsEncryption" -}}
+{{- if (ne .Values.tls.internodeEncryption "none") -}}
+    {{- printf "%s" .Values.tls.internodeEncryption -}}
+{{- else -}}
+    {{- printf "none" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if encryption via TLS should be configured
+*/}}
+{{- define "scylladb.tlsEncryptionEnabled" -}}
+{{- if or .Values.tls.clientEncryption ( ne "none" (include "scylladb.internode.tlsEncryption" . )) -}}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the Scylladb TLS credentials secret
+*/}}
+{{- define "scylladb.tlsSecretName" -}}
+{{- if .Values.tls.existingSecret -}}
+    {{- print (tpl .Values.tls.existingSecret $) -}}
+{{- else -}}
+    {{- printf "%s-crt" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the Scylladb TLS credentials secret
+*/}}
+{{- define "scylladb.tlsCASecretName" -}}
+{{- if .Values.tls.existingCASecret -}}
+    {{- print (tpl .Values.tls.existingCASecret $) -}}
+{{- else -}}
+    {{- printf "%s-ca-crt" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a TLS credentials secret object should be created
+*/}}
+{{- define "scylladb.createTlsSecret" -}}
+{{- if and (include "scylladb.tlsEncryptionEnabled" .) .Values.tls.autoGenerated.enabled (not .Values.tls.existingSecret)  }}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
```

Comment:
Mising`eq .Values.tls.autoGenerated.engine "helm"` condition, right?

Reply:
Not really, as it would also apply with cert-manager

Reply:
Then revisit conditions at `bitnami/scylladb/templates/tls-secret.yaml` since you're creating the TLS secrets using Helm regardless the engine.